### PR TITLE
Kinesis: clean up contexts in kinesisFlowWithContext on failure

### PR DESCRIPTION
**Pull request by hagerf**
Is one possibility for solving contexts that need handling when writing to a Kinesis data stream fails
References #2814

### CODE REVIEW

**Comment by hagerf** [comment]
Not sure why the binary compatibility failed, maybe someone could give some pointers?

File: kinesis/src/main/scala/akka/stream/alpakka/kinesis/scaladsl/KinesisFlow.scala [hunk 1]
```diff
@@ -38,7 +37,11 @@ object KinesisFlow {
       .via(withContext(streamName, settings))
       .map(_._1)
 
-  def withContext[T](streamName: String, settings: KinesisFlowSettings = KinesisFlowSettings.Defaults)(
+  val NoCleanup: Any => Unit = _ => ()
+
+  def withContext[T](streamName: String,
+                     settings: KinesisFlowSettings = KinesisFlowSettings.Defaults,
+                     cleanUpContextOnFailure: T => Unit = NoCleanup)(
```
[the remark this answers is not in the document]
Comment (hagerf) [reply kinesis/src/main/scala/akka/stream/alpakka/kinesis/scaladsl/KinesisFlow.scala]: ok. Have done that now, and a rebase to master

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by ennru** [comment]
Thank you!

**Comment by hagerf** [comment]
Great, thanks!

**Pull request merged**